### PR TITLE
Issue #3283375 by SV: As a LU I want to disable following me for other users

### PR DESCRIPTION
**Pull request by volodymyr-sydor**
## Problem
As a LU I want to disable user following

## Solution
- Added possibility to disable and prevent following for other users

## Issue tracker
- https://www.drupal.org/project/social/issues/3283375
- https://getopensocial.atlassian.net/browse/YANG-7374

## How to test
- [ ] Using the latest version of Open Social with the social_follow_user module enabled
- [ ] As a SM, activate `/admin/config/opensocial/follow-user` user follow
- [ ] As a LU, try to enable `/user/ID/edit` the `Prevent members from following me` option
- [ ] Switch back to SM and go to the info page of the LU
- [ ] When opened page, the follow button is not available anymore

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] If applicable (I.E. new hook_updates) the update path from previous versions (major and minor versions) are tested
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Screenshots
Screenshot-2022-05-31-at-16-44-03-png-2740×1634-

Screenshot-2022-05-31-at-16-53-32-png-2388×1576-

## Release notes
Provides an ability to disable follow feature for other users

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2967-xi5hc1yplc489g6jlevpvqkfzhlujwx8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62961f196eb9e7df1a3aef01

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2967-xi5hc1yplc489g6jlevpvqkfzhlujwx8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62961f196eb9e7df1a3aef01

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2967-xi5hc1yplc489g6jlevpvqkfzhlujwx8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62961f196eb9e7df1a3aef01

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2967-xi5hc1yplc489g6jlevpvqkfzhlujwx8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62961f196eb9e7df1a3aef01

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2967-xi5hc1yplc489g6jlevpvqkfzhlujwx8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62961f196eb9e7df1a3aef01

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2967-xi5hc1yplc489g6jlevpvqkfzhlujwx8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62961f196eb9e7df1a3aef01

File: modules/social_features/social_follow_user/social_follow_user.module [hunk 1]
```diff
@@ -151,7 +154,14 @@ function social_follow_user_theme(): array {
...
 function social_follow_user_preprocess_profile(array &$variables): void {
-  if ($variables['elements']['#view_mode'] === 'teaser') {
+  // Skip any action if "follow user" is not active.
+  if (!\Drupal::config('social_follow_user.settings')->get('status')) {
+    return;
+  }
+
+  $view_mode = $variables['elements']['#view_mode'];
+
+  if ($view_mode === 'teaser') {
```
Comment (ribel) [inline modules/social_features/social_follow_user/social_follow_user.module:164]: @volodymyr-sydor Do we really need to add conditions for specific view modes?
Isn't it better to check for all possible view modes?

File: modules/social_features/social_follow_user/social_follow_user.module [hunk 2]
```diff
@@ -162,9 +172,23 @@ function social_follow_user_preprocess_profile(array &$variables): void {
         $variables['following'] = TRUE;
```
Comment (ribel) [inline modules/social_features/social_follow_user/social_follow_user.module:172]: @volodymyr-sydor I see we already have this `following` variable. Where it is used?
Do we really need extra `disable_following` variable?

File: modules/social_features/social_follow_user/social_follow_user.module [hunk 3]
```diff
@@ -187,3 +211,73 @@ function social_follow_user_preprocess_flag(array &$variables): void {
...
+/**
+ * Implements hook_form_FORM_ID_alter() for user_form().
+ */
+function social_follow_user_form_user_form_alter(array &$form, FormStateInterface $form_state): void {
+  $profile = _social_profile_get_profile_from_route();
+  $active = \Drupal::config('social_follow_user.settings')->get('status');
+
+  if ($profile instanceof ProfileInterface && $active) {
+    $user_data = \Drupal::service('user.data');
+    $disable_following = $user_data->get('social_profile_privacy', (int) $profile->get('uid')->target_id, 'disable_following');
```
Comment (ribel) [inline modules/social_features/social_follow_user/social_follow_user.module:224]: @volodymyr-sydor I would suggest to rename it to `following_enabled`
 or `following_enabled`.

Double negatives are bad in plain English as well as computer code. See explanation [here](https://stackoverflow.com/questions/13470724/naming-convention-isenable-vs-isdisable).

**Review by ribel — CHANGES_REQUESTED** [review]
@volodymyr-sydor I have tested functionality and all works good and as expacted.
But I have suggestion for naming. I know it started from ticket title, but in code I think it would be better to fix it before it merged.

File: modules/social_features/social_follow_user/social_follow_user.module (hunk 1 above)
Comment (volodymyr-sydor) [reply modules/social_features/social_follow_user/social_follow_user.module:164]: We use follow functionality in specific view modes and if we add it then it will be available

File: modules/social_features/social_follow_user/social_follow_user.module (hunk 2 above)
Comment (volodymyr-sydor) [reply modules/social_features/social_follow_user/social_follow_user.module:172]: Yes, because followers/following variables uses to indicate which label we should use on the different followers/following list
but in current case we should have addition flag to check follow privacy rule

File: modules/social_features/social_follow_user/social_follow_user.module (hunk 3 above)
Comment (volodymyr-sydor) [reply modules/social_features/social_follow_user/social_follow_user.module:224]: I agree but then using it for this label seems doesn't match:
Screenshot 2022-06-03 at 09 23 54

File: modules/social_features/social_follow_user/social_follow_user.module (hunk 2 above)
Comment (ribel) [reply modules/social_features/social_follow_user/social_follow_user.module:172]: Ok

File: modules/social_features/social_follow_user/social_follow_user.module (hunk 1 above)
Comment (ribel) [reply modules/social_features/social_follow_user/social_follow_user.module:164]: Ok

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2967-xi5hc1yplc489g6jlevpvqkfzhlujwx8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62961f196eb9e7df1a3aef01

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2967-xi5hc1yplc489g6jlevpvqkfzhlujwx8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62961f196eb9e7df1a3aef01

File: modules/social_features/social_follow_user/social_follow_user.module [hunk 4]
```diff
@@ -187,3 +211,73 @@ function social_follow_user_preprocess_flag(array &$variables): void {
...
+  }
+}
+
+/**
+ * Check & disable following due to privacy settings.
+ *
+ * @param array $variables
+ *   Array with variables of a profile.
+ */
+function _social_follow_user_disable_following(array &$variables): void {
```
[the remark this answers is not in the document]
Comment (volodymyr-sydor) [reply modules/social_features/social_follow_user/social_follow_user.module]: done

File: modules/social_features/social_follow_user/social_follow_user.module (hunk 3 above)
Comment (volodymyr-sydor) [reply modules/social_features/social_follow_user/social_follow_user.module]: done

File: modules/social_features/social_follow_user/social_follow_user.install [hunk 5]
```diff
@@ -149,3 +156,28 @@ function social_follow_user_update_11203(): string {
...
+ * Change a weigh of the social_follow_user module.
+ */
+function social_follow_user_update_11401(): void {
+  module_set_weight('social_follow_user', 6);
+}
+
+/**
+ * Enable following status of all existing users by default.
+ */
+function social_follow_user_update_11402(): void {
```
Comment (ribel) [inline modules/social_features/social_follow_user/social_follow_user.install:170]: @volodymyr-sydor It would be good to avoid this update hook, since ability to follow should be enabled by default for all users. Also on platforms where there are 50k of users it may such update may out of memory, so we would need to do it with batch.

File: modules/social_features/social_follow_user/social_follow_user.module [hunk 6]
```diff
@@ -187,3 +212,80 @@ function social_follow_user_preprocess_flag(array &$variables): void {
...
+    Cache::invalidateTags(['profile:' . $uid]);
+  }
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_insert().
+ *
+ * @throws \Drupal\Core\Entity\EntityStorageException
+ */
+function social_follow_user_user_insert(UserInterface $user): void {
```
Comment (ribel) [inline modules/social_features/social_follow_user/social_follow_user.module:274]: @volodymyr-sydor It would be good to avoid this update hook, since ability to follow should be enabled by default for all users.

File: modules/social_features/social_follow_user/src/Service/SocialFollowUserHelperInterface.php [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+  /**
+   * Check if following is enabled for the profile.
+   *
+   * @param \Drupal\profile\Entity\ProfileInterface $profile
+   *   The profile entity object.
+   *
+   * @return bool
+   *   TRUE or FALSE depending upon following status.
+   */
+  public function isFollowingEnabled(ProfileInterface $profile): bool;
```
Comment (ribel) [inline modules/social_features/social_follow_user/src/Service/SocialFollowUserHelperInterface.php:44]: @volodymyr-sydor Do we really need to have this method? Can we just have `setFollowingStatus()` and `getFollowingStatus()`?

File: modules/social_features/social_follow_user/src/Service/SocialFollowUserHelper.php [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+    $following_enabled = TRUE;
+
+    // Check if disabled user following due to privacy settings.
+    if (!$this->getFollowingStatus((int) $profile->get('uid')->target_id)) {
+      $following_enabled = FALSE;
+
+      // Check if user already followed.
+      /** @var \Drupal\flag\FlagInterface $flag */
+      $flag = $this->flagService->getFlagById('follow_user');
+      // And display only "Unfollow" button because we should leave the ability
```
Comment (ribel) [inline modules/social_features/social_follow_user/src/Service/SocialFollowUserHelper.php:72]: @volodymyr-sydor Do we have it in acceptance criteria that other users should be able to unfollow if following is disabled at all for this user?

**Review by ribel — CHANGES_REQUESTED** [review]
@volodymyr-sydor I still have some questions - please check.
Also please move PR from Draft and merge main branch to fix failing tests due to not applied patch.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2967-xi5hc1yplc489g6jlevpvqkfzhlujwx8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62961f196eb9e7df1a3aef01

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2967-xi5hc1yplc489g6jlevpvqkfzhlujwx8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62961f196eb9e7df1a3aef01

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2967-xi5hc1yplc489g6jlevpvqkfzhlujwx8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62961f196eb9e7df1a3aef01

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2967-xi5hc1yplc489g6jlevpvqkfzhlujwx8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62961f196eb9e7df1a3aef01

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2967-xi5hc1yplc489g6jlevpvqkfzhlujwx8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62961f196eb9e7df1a3aef01

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2967-xi5hc1yplc489g6jlevpvqkfzhlujwx8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62961f196eb9e7df1a3aef01

File: modules/social_features/social_follow_user/social_follow_user.module [hunk 9]
```diff
@@ -187,3 +211,90 @@ function social_follow_user_preprocess_flag(array &$variables): void {
...
+
+  if ($profile instanceof ProfileInterface && $active) {
+    $following_enabled = \Drupal::service('social_follow_user.helper')
+      ->getFollowingStatus((int) $profile->get('uid')->target_id);
+
+    $form['profile_privacy']['social_profile_following_enabled'] = [
+      '#type' => 'checkbox',
+      '#title' => t('Allow members to follow me'),
+      '#default_value' => $following_enabled,
+      '#description' => t('This section allows you to turn off the ability for
```
Comment (ribel) [inline modules/social_features/social_follow_user/social_follow_user.module:230]: @volodymyr-sydor Please change to: _This selection allows you to turn **on** the ability for members to follow your activity and receive notifications about new content you create._

Comment (volodymyr-sydor) [reply modules/social_features/social_follow_user/social_follow_user.module]: done

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2967-xi5hc1yplc489g6jlevpvqkfzhlujwx8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62961f196eb9e7df1a3aef01

File: modules/social_features/social_follow_user/src/Service/SocialFollowUserHelper.php [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+      }
+    }
+    return $following_enabled;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setFollowingStatus(int $uid, $status = TRUE): void {
+    $this->userData->set(
```
Comment (Ressinel) [inline modules/social_features/social_follow_user/src/Service/SocialFollowUserHelper.php:85]: @volodymyr-sydor I more like an approach with a new profile field instead of the user data. Since if we use the user data we need to use batch for hook_install to set the default value and I think it is not good https://github.com/goalgorilla/open_social/pull/2967/files#diff-d3d7b1d9964a18dba55138aef6c9585a65978162753ff93181c2709491e32289R38
But I also understand that many fields for the profile are also not well. So I want to hear your thoughts @navneet0693 @tbsiqueira

cc @ribel

Comment (volodymyr-sydor) [reply modules/social_features/social_follow_user/src/Service/SocialFollowUserHelper.php:85]: @Ressinel Yes, seems like it will be even easier if we create a new profile field, for example, `field_profile_following` then we don't need to update existing users & add any action for new accounts, etc.
@navneet0693 @tbsiqueira are you ok with that approach?

**Review by tbsiqueira — CHANGES_REQUESTED** [review]

File: modules/social_features/social_follow_user/src/Service/SocialFollowUserHelper.php (hunk 10 above)
Comment (volodymyr-sydor) [reply modules/social_features/social_follow_user/src/Service/SocialFollowUserHelper.php:85]: @tbsiqueira 
The acceptance criteria are that `everyone that is already following the user should stay a follower` 
> Will my account still be listed under the "following users" overview from another user? 

Yes, because currently, we disabling only the possibility of following a user. 

If we need to extend functionality and hide followers from the list or etc I would recommend do it separately in follow-up and decide which AC should be applied, because it could make the process review of the task more complicated than was planned. Also, it will require additional testing.  
cc: @ribel

**Comment by volodymyr-sydor** [comment]
> @volodymyr-sydor please also check the user profile page itself, even after I disable the "follow" option in one user, I'm still able to follow it from the "profile" page.
> 
> image

@tbsiqueira 
The button displays in the statistic block, have you applied a theme patch as well? (here is the [issue](https://www.drupal.org/project/socialblue/issues/3283531))

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2967-xi5hc1yplc489g6jlevpvqkfzhlujwx8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62961f196eb9e7df1a3aef01

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2967-xi5hc1yplc489g6jlevpvqkfzhlujwx8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62961f196eb9e7df1a3aef01

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2967-xi5hc1yplc489g6jlevpvqkfzhlujwx8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62961f196eb9e7df1a3aef01

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2967-xi5hc1yplc489g6jlevpvqkfzhlujwx8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62961f196eb9e7df1a3aef01

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2967-xi5hc1yplc489g6jlevpvqkfzhlujwx8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62961f196eb9e7df1a3aef01

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2967-xi5hc1yplc489g6jlevpvqkfzhlujwx8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62961f196eb9e7df1a3aef01

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2967-xi5hc1yplc489g6jlevpvqkfzhlujwx8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62961f196eb9e7df1a3aef01

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2967-xi5hc1yplc489g6jlevpvqkfzhlujwx8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62961f196eb9e7df1a3aef01

**Comment by ribel** [comment]
Hey @Ressinel @tbsiqueira, @volodymyr-sydor implemented requested changes and it looks good to me now. 
Could you check it again and let us know if we can merged it?

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2967-xi5hc1yplc489g6jlevpvqkfzhlujwx8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62961f196eb9e7df1a3aef01

**Review by Ressinel — COMMENTED** [review]
@volodymyr-sydor I have found a case, where the user can follow another one even if the setting `Allow members to follow me`, is disabled:
- `user_1` and `user_2`
- `user_1` has followed `user_2`
- `user_2` set `FALSE` for the `Allow members to follow me` setting
- `user_1` is still able to unfollow `user_2`
- `user_1` press unfollow button and at the same time can follow `user_2` again even if `Allow members to follow me` is `FALSE` for `user_2`

Not sure that it is critical but at least it is not logical that some users can re-follow even if `Allow members to follow me` is `FALSE`.

**Comment by volodymyr-sydor** [comment]
> @volodymyr-sydor I have found a case, where the user can follow another one even if the setting `Allow members to follow me`, is disabled:
> 
> * `user_1` and `user_2`
> * `user_1` has followed `user_2`
> * `user_2` set `FALSE` for the `Allow members to follow me` setting
> * `user_1` is still able to unfollow `user_2`
> * `user_1` press unfollow button and at the same time can follow `user_2` again even if `Allow members to follow me` is `FALSE` for `user_2`
> 
> Not sure that it is critical but at least it is not logical that some users can re-follow even if `Allow members to follow me` is `FALSE`.

@Ressinel 
We've left the possibility to unfollow the user. If it's caused a new issue I would suggest do not show "unfollow" button as well. So, I will remove it as it even wasn't in the acceptance criteria

**Comment by Ressinel** [comment]
> > @volodymyr-sydor I have found a case, where the user can follow another one even if the setting `Allow members to follow me`, is disabled:
> > 
> > * `user_1` and `user_2`
> > * `user_1` has followed `user_2`
> > * `user_2` set `FALSE` for the `Allow members to follow me` setting
> > * `user_1` is still able to unfollow `user_2`
> > * `user_1` press unfollow button and at the same time can follow `user_2` again even if `Allow members to follow me` is `FALSE` for `user_2`
> > 
> > Not sure that it is critical but at least it is not logical that some users can re-follow even if `Allow members to follow me` is `FALSE`.
> 
> @Ressinel We've left the possibility to unfollow the user. If it's caused a new issue I would suggest do not show "unfollow" button as well. So, I will remove it as it even wasn't in the acceptance criteria

@volodymyr-sydor I think we should leave the possibility to unfollow in this case. At least it sounds logical.
@tbsiqueira @ribel need your thoughts as well

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2967-xi5hc1yplc489g6jlevpvqkfzhlujwx8.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62961f196eb9e7df1a3aef01

**Comment by volodymyr-sydor** [comment]
> @Ressinel As @volodymyr-sydor said adding possibility to unfollow even if you disable follow was already out of the scope of this story. But since it also looked logical to us we implemented it. But then we come to the use case which you described. And I agree that it would be good to fix that, but not sure how much time it will take and we already spent more time that planned on it.
> 
> @volodymyr-sydor maybe you can still investigate it a bit and besides that we also now need to fix conflict in composer after recent merges.

I've adjusted the access check and now after "unfollow" action the button is not available for user with disabled "following" option

File: modules/social_features/social_follow_user/social_follow_user.module [hunk 11]
```diff
@@ -162,7 +172,21 @@ function social_follow_user_preprocess_profile(array &$variables): void {
...
+  if ($view_mode === 'preview') {
+    _social_follow_user_check_following_status($variables);
+  }
+}
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function social_follow_user_preprocess_profile__profile__statistic(array &$variables): void {
+  if (!\Drupal::config('social_follow_user.settings')->get('status')) {
```
[the remark this answers is not in the document]
Comment (volodymyr-sydor) [reply modules/social_features/social_follow_user/social_follow_user.module:193]: done

**Review by Ressinel — APPROVED** [review]
Great work! Looks good to me!

**Review by tbsiqueira — APPROVED** [review]

**Comment by Ressinel** [comment]
🍒-picked to `11.4.x`
